docs/quickstart.rst

```diff
@@ -61,198 +199,276 @@ instance had already existing in the destination directory, this command
-        Name |   Port | Group |  Node |     Current State |    Assigned State
-   ----------+--------+-------+-------+-------------------+------------------
-   127.0.0.1 |   6010 |     0 |     1 |           primary |           primary
-   127.0.0.1 |   6011 |     0 |     2 |         secondary |         secondary
+   ssh -l ha-admin `vm_ip monitor` pg_autoctl show state --pgdata monitor
 
-This looks good. We can add data to the primary, and watch it get
-reflected in the secondary.
+                              Name |   Port | Group |  Node |     Current State |    Assigned State
+   --------------------------------+--------+-------+-------+-------------------+------------------
+   ha-demo-a.internal.cloudapp.net |   5432 |     0 |     1 |           primary |           primary
+   ha-demo-b.internal.cloudapp.net |   5432 |     0 |     2 |         secondary |         secondary
+
+
+This looks good. We can add data to the primary, and later see it appear in the
+secondary. We'll connect to the database from inside our "app" virtual machine,
+using a connection string obtained from the monitor. First we'll get the
+connection string and store it in a local environment variable:
 
 .. code-block:: bash
 
-   # add data to primary
-   psql -p 6010 \
-     -c 'create table foo as select generate_series(1,1000000) bar;'
+   APP_DB_URI=$( \
+     ssh -l ha-admin `vm_ip monitor` \
+       pg_autoctl show uri --formation default --pgdata monitor \
+   )
 
-   # query secondary
-   psql -p 6011 -c 'select count(*) from foo;'
-     count
-   ---------
-    1000000
+The connection string contains both our nodes, comma separated, and includes
+the url parameter ``?target_session_attrs=read-write`` telling psql that we
+want to connect to whichever of these servers supports reads *and* writes.
+That will be the primary server.
+
+.. code-block:: bash
+
+   # connect to database via psql on the app vm and
+   # create a table with a million rows
+   ssh -l ha-admin -t `vm_ip app` -- \
+     psql "\"$APP_DB_URI\"" \
+       -c "\"CREATE TABLE foo AS SELECT generate_series(1,1000000) bar;\""
 
 Cause a failover
 ----------------
 
-This plot is too boring, time to introduce a problem. We’ll run the
-primary out of disk space and watch the secondary get promoted.
-
-In one terminal let’s keep an eye on events:
+Now that we've added data to node A, let's switch which is considered
+the primary and which the secondary. After the switch we'll connect again
+and query the data, this time from node B.
 
 .. code-block:: bash
 
-   watch pg_autoctl show events --pgdata ./monitor
+   # initiate failover to node B
+   ssh -l ha-admin -t `vm_ip monitor` \
+     pg_autoctl perform switchover --pgdata monitor
 
-In another terminal we’ll consume node A’s disk space and try to restart
-the database. It will refuse to start up.
+   # watch the progress
+   watch -d \
+     ssh -l ha-admin `vm_ip monitor` pg_autoctl show state --pgdata monitor
```

Comment:
Better use watch on the remote instance rather than reconnect using ssh every other second, I think:

```
ssh -t -l ha-admin `vm_ip monitor` -- watch -n 1 pg_autoctl show state --pgdata monitor
```

Reply:
Also this fails for me on node a when catching-up again with the following, which I need to understand if it's a bug in 1.2 (that was a hasty release) or if we still have it, and then we need to fix it of course.

```
pg_basebackup: could not connect to server: FATAL:  no pg_hba.conf entry for replication connection from host "10.0.1.7", user "pgautofailover_replicator", SSL on
```

Reply:
So that's because we have two reserve DNS host names for a given IP address and Postgres is confused, it works better without the extra DNS settings.